src/effects/backends/builtin/whitenoiseeffect.cpp

```diff
@@ -50,24 +58,40 @@ void WhiteNoiseEffect::processChannel(
 
     WhiteNoiseGroupState& gs = *pState;
 
-    CSAMPLE drywet = static_cast<CSAMPLE>(m_pDryWetParameter->value());
+    const CSAMPLE drywet = static_cast<CSAMPLE>(m_pDryWetParameter->value());
```

Comment:
Is this also a gain value? 
```suggestion
    const CSAMPLE_GAIN drywet = static_cast<CSAMPLE_GAIN>(m_pDryWetParameter->value());
```

Reply:
Well, `CSAMPLE_GAIN` is just a typedef alias for `CSAMPLE`... I don't really know what the purpose of the `CSAMPLE` type is at all.

Reply:
The types are explained here: https://github.com/mixxxdj/mixxx/blob/main/src/util/types.h
Does this solve your questions, or do we need to improve the description?

Reply:
Thanks. I only have one question and a couple remarks:
What does the "POT" stand for in "POT array indices" (`SINT` description)?
`SAMPLE` (16-bit) seems virtually unused in our codebase, why do we have it in the first place?
`CSAMPLE_GAIN` should probably a new type-kinda thing with compiletime checks (and in debug-builds) to ensure that the range-invariant is upheld. Maybe something similar for `CSAMPLE` as long as we can ensure they stay zero-cost during runtime.

Reply:
Oh that's a typo 
It should be POD https://en.cppreference.com/w/cpp/named_req/PODType

Reply:
> SAMPLE (16-bit) seems virtually unused in our codebase, why do we have it in the first place?

This is legacy. In early versions of Mixxx Soundsource have produced buffers of SAMPLE. We have changed that at one point. 

For now  CSAMPLE and CSAMPLE_GAIN are only for readability purpose, a lot of room for improvements. I guess all these is a topic of refactoring when we introduce n-channel

Reply:
Thanks for the explanations